File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/ItemBulkEnum.java
package ai.ragu.charactermanager.enumeration;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.HashMap;
import java.util.Map;

public enum ItemBulkEnum implements WeaponTraitEnum {
    SMALL("small"),
    MEDIUM("medium"),
    LARGE("large"),
    TWO("2");

    private final String value;
    private final static Map<String, ItemBulkEnum> CONSTANTS = new HashMap<>();

    static {
        for (ItemBulkEnum c: values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    ItemBulkEnum(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

    @JsonCreator
    public static ItemBulkEnum fromValue(String value) {
        ItemBulkEnum constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(value);
        } else {
            return constant;
        }
    }
}

File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/ItemBulkEnum.kt
package ai.ragu.charactermanager.enumeration

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue
import java.util.*

enum class ItemBulkEnum(private val value: String) : WeaponTraitEnum {
    SMALL("S"), MEDIUM("M"), LARGE("L"), TWO("2");

    companion object {
        private val CONSTANTS: MutableMap<String, ItemBulkEnum> = HashMap()
        @JsonCreator
        fun fromValue(value: String?): ItemBulkEnum {
            val constant = CONSTANTS[value]
            return constant ?: throw IllegalArgumentException(value)
        }

        init {
            for (c in values()) {
                CONSTANTS[ai.ragu.charactermanager.enumeration.c.value] = ai.ragu.charactermanager.enumeration.c
            }
        }
    }

    override fun toString(): String {
        return value
    }

    @JsonValue
    fun value(): String {
        return value
    }

}